### PR TITLE
Add Unlock for TestHammerController

### LINKED ISSUE #34936: Miss to call Unlock in TestHammerController
**Kubernetes version** (use `kubectl version`):
The latest

**What you expected to happen**:
when call Lock(), we need to call Unlock()

I have send a PR to fix this issue and PTAL.

Signed-off-by: yuexiao-wang wang.yuexiao@zte.com.cn

### PR DESCRIPTION
**Pull request by yuexiao-wang**
**What this PR does / why we need it**:
Add Unlock for TestHammerController

**Which issue this PR fixes** _(optional, in `fixes #<issue number>(, #<issue_number>, ...)` format, will close that issue when PR gets merged)_: 
Fixes #34936

Signed-off-by: yuexiao-wang wang.yuexiao@zte.com.cn

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34937)

### CODE REVIEW

**Comment by yuexiao-wang** [comment]
@lavalamp PTAL
Thanks!

**Comment by yuexiao-wang** [comment]
@lavalamp Could you give me some suggestions?
Thanks!

File: pkg/client/cache/controller_test.go [hunk 1]
```diff
@@ -284,6 +284,7 @@ func TestHammerController(t *testing.T) {
 	close(stop)
 
 	outputSetLock.Lock()
+	defer outputSetLock.Unlock()
```
Comment (lavalamp) [inline pkg/client/cache/controller_test.go]: Sorry for the delay, I have been on paternity leave...

I suspect the original reason for omitting this is the TODO on line 282 above. I.e., it's purposely locking out any goroutines still running so they don't affect subsequent tests.

Comment (yuexiao-wang) [inline pkg/client/cache/controller_test.go]: Ok, I will close this PR.

**Pull request closed without merging**